raster-foundry/raster-foundry
# Don't rewrite requests for map token auth

## Overview

This PR does away with request rewriting for map token auth, opting instead for another authenticator. It's way less complicated and not noticeably less DRY.

### Checklist

- ~Description of PR is in an appropriate section of the [changelog](https://github.com/raster-foundry/raster-foundry/blob/develop/CHANGELOG.md) and grouped with similar changes if possible~

### Notes

How did this break within 24 hours of my opening the first map token auth pr? Because when I opened that one I was only doing GetCapabilities. Rewriting the request and calling `routes` on it affected the result of calling `getServiceUrl` on the request, which made it so that `DescribeCoverage` / `GetCoverage` / `GetMap` were being routed to different places from where the `GetCapabilities` requests were going. I couldn't test that at first, because those routes weren't implemented.

## Testing Instructions

 * share a private project with a map token
 * add `/wcs/<project id>/map-token/<map token id>` as a wcs layer in qgis and confirm you can view one of the layers
 * add `/wms/<project id>/map-token/<map token id>` as a wms/wmts layer in qgis and confirm you can view one of the layers

Closes #4775 (for real this time)
